pkg/component/http.go

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"context"
+	"net/http"
+	"time"
+)
+
+const (
+	// defaultTimeout is the default timeout for the HTTP client.
+	defaultTimeout = 10 * time.Second
+)
+
+// HTTPClientOpt applies an option to the HTTP client.
+type HTTPClientOpt func(*http.Client)
+
+// WithTimeout sets the HTTP client timeout.
+func WithTimeout(t time.Duration) HTTPClientOpt {
```

Comment:
Not sure if I like this.

First of all, this becomes `component.WithTimeout`, which conflicts with actual component options. 
Second, I think it's a bit of an anti-pattern to override this everywhere - We added `HTTPClient` in order to have uniform HTTP client settings everywhere. I'd prefer that this option is used only in tests at most.

Reply:
We should probably also check how this interacts with the context deadline that we use in individual requests.